[OSDOCS-17885] Corrected Cloud Expert Tutorials for OSD for CQA jobs (#108735)
openshift/openshift-docs: modules/cloud-experts-osd-update-component-routes-create-tls-certificates.adoc

:_mod-docs-content-type: PROCEDURE
[id="cloud-experts-osd-update-component-routes-create-tls-certificates_{context}"]
= Creating TLS certificates for each component route

[role="_abstract"]
Create three self-signed certificates, one for each component route. Trust them on your system so you can open each new hostname in a browser.

[WARNING]
====

> Use this flow for learning only, not for production. For live systems, ask your certificate authority (CA) how to get real certificates.

Reviewer: ```suggestion
Use this flow for learning only, not for production. For live systems, request valid (production-ready?) certificates from your certificate authority (CA).
```

[WARNING]
====
Use this flow for learning only, not for production. For live systems, request valid ceritificates from your certificate authority (CA).
====

[IMPORTANT]
====
Use one certificate per route to prevent issues with HTTP/2 connection coalescing. Wildcard certificates and subject alternative names (SAN) certificates are not supported.
====

This example uses the following custom component routes:
+
* `console.my-new-domain.dev` for Console
* `downloads.my-new-domain.dev` for Downloads
* `oauth.my-new-domain.dev` for OAuth

.Procedure
* For each route, run the example `openssl` commands. Set `-subj` to that route's domain name:
+
.Example output:
[source,bash]
----
$ openssl req -newkey rsa:2048 -new -nodes -x509 -days 365 -keyout key-console.pem -out cert-console.pem -subj "/CN=console.my-new-domain.dev"
$ openssl req -newkey rsa:2048 -new -nodes -x509 -days 365 -keyout key-downloads.pem -out cert-downloads.pem -subj "/CN=downloads.my-new-domain.dev"
$ openssl req -newkey rsa:2048 -new -nodes -x509 -days 365 -keyout key-oauth.pem -out cert-oauth.pem -subj "/CN=oauth.my-new-domain.dev"
----

.Verification

> * Check that the `.pem` certificate and key files exist:

Author: I need to confirm this is a useful/valid command.
